### PR TITLE
feat: add app for slow-restart

### PR DESCRIPTION
**Pull request by matbur**

### CODE REVIEW

File: slow-restart/README.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+linkerd-proxy-injector from Control Plane.
+
+## Example scenario:
+
+Imagine that you upgraded Linkerd Control Plane from version edge-21.9.1 to stable-2.11.0. All your Data Planes are
+still running with the old version. In order to change this you would need to restart all running deployments. If you
+know k8s, you might know that to do this you could just run `kubectl rollout restart deployment <your-deployment-name>`
+which would restart <your-deployment-name>.
+
+You could also run ` kubectl rollout restart deployment`, but this command restarts all deployment instantly.
```
Comment (arekziobrowski) [inline slow-restart/README.md:13]:
```suggestion
You could also run `kubectl rollout restart deployment`, but this command restarts all deployment instantly.
```

File: slow-restart/main.go [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+var (
+	timeout = flag.DurationP("timeout", "t", 10*time.Minute, "how long wait for deleting pods")
+	sleep   = flag.DurationP("sleep", "s", time.Minute, "how log wait between each deployment restart")
```
Comment (arekziobrowski) [inline slow-restart/main.go:25]:
```suggestion
	sleep   = flag.DurationP("sleep", "s", time.Minute, "how long to wait between each deployment restart")
```

File: slow-restart/main.go [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+		if err := waitForDeletePods(ctx, podNames, *timeout); err != nil {
+			log.Fatal(err)
+		}
+
+		time.Sleep(*sleep)
+	}
+}
+
+func getKubernetesCli() (*kubernetes.Clientset, error) {
+	kubeCfgPath := filepath.Join(os.Getenv("HOME"), ".kube", "config")
```
Comment (arekziobrowski) [inline slow-restart/main.go:69]: Maybe we should first check the `KUBECONFIG` environment variable and if it is empty fall back to `$HOME/.kube/config`?

https://kubernetes.io/docs/tasks/access-application-cluster/configure-access-multiple-clusters/#append-home-kube-config-to-your-kubeconfig-environment-variable

File: slow-restart/README.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
+# slow-restart
```
Comment (mkrutik) [inline slow-restart/README.md:1]: I would rather name it smth like `linkerd-dataplane-updater`

File: slow-restart/README.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+Script slowly restarts all deployments in `default` namespace for which linkerd-proxy version differs with
+linkerd-proxy-injector from Control Plane.
```
Comment (mkrutik) [inline slow-restart/README.md:4]:
```suggestion
```suggestion
Script restarts all deployments one-by-one from `default` namespace for which `linkerd-proxy` version differs with
`linkerd-proxy-injector` from Control Plane.
```

File: slow-restart/README.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
+# slow-restart
+
+Script slowly restarts all deployments in `default` namespace for which linkerd-proxy version differs with
```
Comment (arekziobrowski) [inline slow-restart/README.md:3]:
```suggestion
Script slowly restarts all deployments in `default` namespace for which linkerd-proxy version differs from
```

**Review by mkrutik — APPROVED** [review]
🔥

File: slow-restart/README.md (hunk 5 above)
Comment (arekziobrowski) [inline slow-restart/README.md:4]:
```suggestion
linkerd-proxy-injector from Control Plane. It aids the process of restarting all running deployments by minimizing the need for i.a. scaling while the restart of dependent services is in progress.
```

File: slow-restart/README.md [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+In this scenario this app comes in handy. It checks which deployments require restarting and restarts them one-by-one
+with additional sleep between if wanted.
+
+## Usage:
+
+```shell
+$ slow-restart -h
+
+Usage of slow-restart:
+  -s, --sleep duration     how log wait between each deployment restart (default 1m0s)
```
Comment (arekziobrowski) [inline slow-restart/README.md:27]:
```suggestion
  -s, --sleep duration     how long to wait between each deployment restart (default 1m0s)
```

File: slow-restart/README.md [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+  -t, --timeout duration   how long wait for deleting pods (default 10m0s)
```
Comment (arekziobrowski) [inline slow-restart/README.md:28]:
```suggestion
  -t, --timeout duration   how long to wait for deleting pods (default 10m0s)
```

**Review by arekziobrowski — APPROVED** [review]
Nice 💯 
Left some minor comments

File: slow-restart/README.md (hunk 4 above)
Comment (matbur) [reply slow-restart/README.md:1]: I'll go with `linkerd-dataplane-updater`

**Pull request merged**